### ISSUE TITLE
[Request]: Skeleton esp in esp

### ISSUE THREAD

**Issue by pierrelasse** [YimMenu/YimMenu#2183]
### Problem

The boxes are weird and are "buggy" when you are near the player.
Also the box does not let you know how the player is moving.

### Solution

Add a skeleton esp where you see the player's bones.
Here are some images i found online:
![image](https://github.com/YimMenu/YimMenu/assets/76223390/ed70f4cb-b6bc-4195-b00d-e5fe07679e9b)
![image](https://github.com/YimMenu/YimMenu/assets/76223390/691a5405-327e-45f9-b54e-9ede07985d61)

### Reason

Remove this field

### Additional context

Ez skid from somewhere?

**Comment by alyxme** [comment]
I would add this or the existing ESP option for NPCs too.

**Comment by pierrelasse** [comment]
> I would add this or the existing ESP option for NPCs too.

So that npcs also have esp?

**Comment by alyxme** [comment]
Not at the current release yet, but it possible to add it to them. This could be a good feature.

**Comment by lonelybud** [comment]
I dont see any crucial requirement. Looks like some fun feature only.

**Comment by pierrelasse** [comment]
I dont see any curcial requirements in many featrues of the menu… And i think that many would use this esp instead of the 2d one.

**Comment by lonelybud** [comment]
> I dont see any curcial requirements in many featrues of the menu… And i think that many would use this esp instead of the 2d one.

yeah one can alter the existing esp or add a checkbox to switch to this one (Basically adding sub feature to existing feature).

**Comment by pierrelasse** [comment]
So, i made it now myself but the problem is that the lines are not rendering above entities.
Does any one know how i can set the lines to render after entities and things like that?
I'm using this native for rendering the lines [0x6B7256074AE34680](https://docs.fivem.net/natives/?_0x6B7256074AE34680=)

**Comment by xiaoxiao921** [comment]
Can you just use the draw techniques used by the current esp and context menu? Through the imgui draw list

**Comment by lonelybud** [comment]
if you need to draw lines from bone to bone then i guess you will need position of bones using these method? - 
```
GET_WORLD_POSITION_OF_ENTITY_BONE
GET_ENTITY_BONE_POSTION
GET_ENTITY_BONE_OBJECT_POSTION
```

**Comment by lonelybud** [comment]
> if you need to draw lines from bone to bone then i guess you will need position of bones using these method? -
> 
> ```
> GET_WORLD_POSITION_OF_ENTITY_BONE
> GET_ENTITY_BONE_POSTION
> GET_ENTITY_BONE_OBJECT_POSTION
> ```

you can ofcourse look into context menu code.  check one of the methods and try to draw one single line.
You can get bone int value from - 

```
ENTITY::GET_ENTITY_BONE_INDEX_BY_NAME(entity, "BONETAG_HEAD");
ENTITY::GET_ENTITY_BONE_INDEX_BY_NAME(entity, "BONETAG_PELVIS");
```

For quick render you can do something like this -
https://forums.gta5-mods.com/topic/30068/get-dimensions-of-entity-like-cars/3

Then for final code you can decide what to use

**Comment by pierrelasse** [comment]
It is already working and drawing everything where it needs to draw. So i already get the bone coords etc. The problem is that the line can get covered by entities.
It would be cool if someone could provide an example with imgui but i dont know how well this will work with coordinates and 3d.

**Comment by xiaoxiao921** [comment]
Check the context menu it draw a rectangle around objects, it's 3d

**Comment by lonelybud** [comment]
> It is already working and drawing everything where it needs to draw. So i already get the bone coords etc. The problem is that the line can get covered by entities. It would be cool if someone could provide an example with imgui but i dont know how well this will work with coordinates and 3d.

are you using GRAPHICS. try imgui it draws on screen i guess and not in game itself?

**Comment by pierrelasse** [comment]
Okay, i will try this later then.

**Comment by lonelybud** [comment]
you should try imgui code in `src\services\context_menu\context_menu_service.cpp` it does get entity dimention and then scale it according to distance.
